crates/turborepo-signals/src/listeners.rs

```diff
@@ -0,0 +1,33 @@
+use futures::{stream, Stream};
+
+use crate::signals::Signal;
+
+#[cfg(windows)]
+/// A listener for Windows Console Ctrl-C events
+pub fn get_signal() -> Result<impl Stream<Item = Option<Signal>>, std::io::Error> {
```

Comment:
Minor detail, but maybe we should make this a custom error type so we can annotate the `run::Error` type with `#[from]` and avoid having to use `map_err` when calling this?